### ISSUE TITLE
How to use the saved model?

### ISSUE THREAD

**Issue by neenaloysius** [ugnelis/tensorflow-rnn-ctc#21]
Got this info. at the end of the training:
2020-01-16 04:49:00,021 - root - INFO - Model saved in file: ./models/model.ckpt
I0116 04:49:00.021643 140352211462016 train.py:241] Model saved in file: ./models/model.ckpt

How to use the saved model for testing a .wav file? Is the code for testing the model already available?

**Comment by ugnelis** [comment]
Hi @neenaloysius, currently there is only `demo.py`, which takes all `test set` and gives output based on trained module. For your described particular use there is no example in this repo. When I will have time, I will make it but I can't promise when. Feel free to make a Pull Request if you will figure out it :)

**Comment by neenaloysius** [comment]
Thanks for your prompt response. Do you have sample code or links to any alignment done (frame state alignment or gloss label alignment) on videos?



 

    On Thursday, January 16, 2020, 01:10:11 PM GMT+5:30, Ugnius Malūkas <notifications@github.com> wrote:  
 
 
Hi @neenaloysius, currently there is only demo.py, which takes all test set and gives output based on trained module. For your described particular use there is no example in this repo. When I will have time, I will make it but I can't promise when. Feel free to make a Pull Request if you will figure out it :)

**Comment by ugnelis** [comment]
I don't have any video. If I understood you correctly, you want to have alignment procedure: `.wav` file -> some code -> transcription, is that right? If yes, I can only provide this code when I will have time. 

If you need now, I can write you how to achieve this:
1. Create `read_audio_file(path)` function in `utils.py`.
2. Copy most of the code from `demo.py` to `some-your-file.py`. It is needed for TensorFlow model checkpoints restoring (at least it was necessary before TensorFlow 2.0)
3. Replace line `test_sequence_lengths = utils.get_sequence_lengths(test_inputs)` with `test_sequence_lengths = utils.get_sequence_lengths([array of single entry which is wav])`. You will probably will need in some more places in `demo.py`.
4. In 
`test_feed = {inputs_placeholder: test_inputs,
 sequence_length_placeholder: test_sequence_lengths}`

Replace `inputs_placeholder: [array with one input]`.

I hope it gives you some help

**Comment by neenaloysius** [comment]
Thanks a lot, sir.
